README.md

```diff
@@ -2124,13 +2124,96 @@ the build will fail if test coverage is below 100%.
 
 You should be able to do the typical stuff:
 
+### 1. Cloning repository:
+Fork the repository and run this(Only for SSH):
+```
+git clone git@github.com:${your_github_username}/foobara.git
+```
+Create a new branch for you to push into
+```
+git checkout -b <branch-name>
+```
+Now navigate to project directory
+
 ```
-git clone git@github.com:foobara/foobara
 cd foobara
+```
+
+### 2. Installing mise (LINUX/WSL)
+Mise is a package manager and helps with managing different versions of ruby. It allows you to switch different versions of ruby.
+
+The installation docs might be updated by mise so here's the reference to that: [Installation Docs](https://mise.jdx.dev/installing-mise.html)
+
+***If you already installed by following mise's docs then, you can skip this section***
+
+Else to setup mise follow the script given below:
+
+```
+sudo apt update -y && sudo apt install -y gpg wget \
+build-essential \
+  libssl-dev zlib1g-dev libreadline-dev libyaml-dev libxml2-dev \
+  libxslt1-dev libffi-dev libgdbm-dev autoconf bison
+wget -qO - https://mise.jdx.dev/gpg-key.pub | gpg --dearmor | sudo tee /etc/apt/keyrings/mise-archive-keyring.gpg 1> /dev/null
+echo "deb [signed-by=/etc/apt/keyrings/mise-archive-keyring.gpg arch=amd64] https://mise.jdx.dev/deb stable main" | sudo tee /etc/apt/sources.list.d/mise.list
+sudo apt update
+sudo apt install -y mise
+```
+
+Verify if it's installed or not by running:
+```
+mise -v
+```
+It should print out an ascii-art saying "mise-en-place"
+### 3. Install required ruby version
+To automatically install the required version mentioned in the .ruby-version in the project's directory, run this command
+
+```
+mise settings add idiomatic_version_file_enable_tools ruby
+```
+
+### 4. Activating mise
+We can activate mise so that it will update the environment variables such that we will use the correct version of ruby.
+```
+echo 'eval "$(mise activate bash)"' >> ~/.bashrc
+```
+Restart the terminal
+
+Now environment variables are updated and you can verify it by running:
+```
+ruby -v
+```
+It will print out the ruby version, which is mentioned in .ruby-version file
+
+Bundler is also necessary so check if it is installed by running:
```

Comment:
I think all Rubies >= 2.6 include bundler so we can delete this since the project requires >= 3.4.